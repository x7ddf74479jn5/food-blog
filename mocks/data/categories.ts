export const categories = {
  rice: { id: "1", slug: "rice", name: "ご飯物", color: "grey" },
  salad: { id: "2", slug: "salad", name: "サラダ", color: "grey" },
  tips: { id: "3", slug: "tips", name: "料理基礎", color: "grey" },
};
