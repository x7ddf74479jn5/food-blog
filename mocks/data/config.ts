import type { TConfig } from "@/types/models/config";

import { dateCommon } from "./utils";

export const mockConfig: TConfig = {
  ...dateCommon,
  apiHost: "https://food-blog.microcms.io/api/v1/",
  host: "https://food-blog-chi.vercel.app/",
  siteTitle: "Food Blog",
  siteDescription: "料理レシピを紹介するブログ",
  siteKeywords: ["料理"],
  siteImage: { url: "https://food-blog-chi.vercel.app/site_image.jpg", width: 100, height: 100 },
  organization: "Pandashark",
  officialSite: "https://next-portfolio-livid.vercel.app/",
  twitterId: "@pandashark6",
  email: "x7ddf74479jn5@gmail.com",
};
