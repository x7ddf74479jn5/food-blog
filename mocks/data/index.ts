export { mockArticles } from "./articles";
export { mockCategories } from "./categories";
export { mockConfig } from "./config";
export { mockPickup } from "./pickup";
export { mockTags } from "./tags";
export { dateCommon } from "./utils";
export { mockWriters } from "./writers";
