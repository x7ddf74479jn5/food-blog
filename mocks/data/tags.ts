import type { TTag } from "@/types";

import { dateCommon } from "./utils";

type TTagCollection = {
  [key: string]: TTag;
};

export const mockTags: TTagCollection = {
  komatsuna: { ...dateCommon, id: "1", slug: "komatshna", name: "小松菜", color: "green" as const },
  mozzarella: { ...dateCommon, id: "2", slug: "mozzarella", name: "モッツァレラ", color: "pink" as const },
  lettuce: { ...dateCommon, id: "3", slug: "lettuce", name: "レタス", color: "green" as const },
  mincedMeat: { ...dateCommon, id: "4", slug: "minced_meat", name: "挽き肉", color: "red" as const },
  rice: { ...dateCommon, id: "5", slug: "rice", name: "ごはん", color: "yellow" as const },
  aburaage: { ...dateCommon, id: "6", slug: "aburaage", name: "油揚げ", color: "gray" as const },
  onion: { ...dateCommon, id: "7", slug: "onion", name: "玉ねぎ", color: "green" as const },
  udon: { ...dateCommon, id: "8", slug: "udon", name: "うどん", color: "yellow" as const },
  misoSoup: { ...dateCommon, id: "9", slug: "miso_soup", name: "味噌汁", color: "gray" as const },
  tomato: { ...dateCommon, id: "10", slug: "tomato", name: "トマト", color: "green" as const },
  preparation: { ...dateCommon, id: "11", slug: "preparation", name: "仕込み", color: "gray" as const },
  ooba: { ...dateCommon, id: "12", slug: "ooba", name: "大葉", color: "green" as const },
};
