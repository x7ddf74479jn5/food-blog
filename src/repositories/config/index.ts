import { cache } from "react";

import { client } from "@/lib/microcms";
import type { TConfig } from "@/types";

export const fetchConfig = cache(async () => {
  const data = await client.get<TConfig>({ endpoint: "configs" });
  return data;
});
