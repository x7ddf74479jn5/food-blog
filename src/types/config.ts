import type { TDateCommon, TImage } from "@/types/utils";

export type TConfig = {
  host: string;
  apiHost: string;
  siteTitle: string;
  siteDescription: string;
  siteKeywords: string[];
  siteImage: TImage;
  twitterId: string;
  organization: string;
} & TDateCommon;
