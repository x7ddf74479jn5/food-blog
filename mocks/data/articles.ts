import type { TArticle } from "@/types";

import { categories } from "./categories";
import { tags } from "./tags";
import { dateCommon } from "./utils";
import { writers } from "./writers";

const articleStock = {
  id: "1",
  title: "基本の一番だしの作り方",
  writer: writers.pandashark,
  description:
    "だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。",
  image: { url: "/images/5026416_s.jpg", height: 427, width: 640 },
  excerpt:
    "だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。",
  body: `**分量： 800ml**

だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。

## 材料
- 水：1000ml
- だし昆布：10g
- かつお削り節：40g

## 作り方
- 固く絞った布巾で、昆布の表面の汚れを軽く拭き取ります。
- 鍋に水と昆布を入れます。
- 弱火で７分ほど熱し、鍋底から小さい泡が出てきたら昆布を取り出します。
- 昆布だしを一旦沸騰させてから、削り節を入れ、弱火で１分煮出します。
- ボウルの上にザルを置き、その上にキッチンペーパーを敷いて出汁を濾します。

---

**二番だしの取り方**
1. 鍋に 500ml の水を入れて沸騰させます。
2. 鍋に一番だしのだしがらを入れ、３分煮出します。
3. 一番だしと同じ方法で濾します。

---

## 保存方法
使わない分は保存容器に移して、冷蔵庫で保存してください。冷蔵保存で 2〜3 日は持ちます。

##  ポイント
だし昆布の表面の白い粉は「うま味」成分なので、取り除かずにそのままお使いください。`,
  category: categories.rice,
  tags: [tags.preparation, tags.misoSoup, tags.udon],
  imageOption: null,
};

type TArticleCollection = {
  [key: string]: TArticle;
};

export const articles: TArticleCollection = {
  stock: { ...dateCommon, ...articleStock },
};
