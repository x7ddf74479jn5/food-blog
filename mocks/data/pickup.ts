import { mockArticles } from "@mocks/data";
import { dateCommon } from "@mocks/data/utils";

export const mockPickup = {
  id: "1",
  name: "旬の食材を使ったおすすめレシピ",
  description: "旬の食材を使ったおすすめレシピ",
  startDate: new Date("2021/10/08 00:00"),
  endDate: new Date("2023/02/08 00:00"),
  articles: Object.values(mockArticles),
  ...dateCommon,
};
