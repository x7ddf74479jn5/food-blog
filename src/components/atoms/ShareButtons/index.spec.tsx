import { mockConfig } from "@mocks/data/config";
import renderer from "react-test-renderer";

import { ShareButtons } from ".";

describe("components/atoms/ShareButtons", () => {
  const { host, siteTitle } = mockConfig;
  it("snapshot", () => {
    const tree = renderer.create(<ShareButtons url={host} title={siteTitle} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
