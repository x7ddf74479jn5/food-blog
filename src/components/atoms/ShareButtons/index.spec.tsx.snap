// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/ShareButtons snapshot 1`] = `
<div
  className="flex flex-row gap-4 justify-center items-center"
>
  <button
    aria-label="twitter"
    className="react-share__ShareButton"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "border": "none",
        "color": "inherit",
        "cursor": "pointer",
        "font": "inherit",
        "padding": 0,
      }
    }
  >
    <div
      className="group inline-block relative cursor-pointer"
    >
      <svg
        className="text-gray-400 hover:text-[#1DA1F2]"
        fill="currentColor"
        height={24}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
        />
      </svg>
      <div
        className="inline-block absolute z-10 text-white bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in -bottom-4 h-3 w-3 rotate-45"
      />
      <div
        className="inline-block absolute z-10 text-white bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in -bottom-8 p-1 text-xs leading-snug}"
      >
        Twitterでシェア
      </div>
    </div>
  </button>
  <button
    aria-label="facebook"
    className="react-share__ShareButton"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "border": "none",
        "color": "inherit",
        "cursor": "pointer",
        "font": "inherit",
        "padding": 0,
      }
    }
    title="Food Blog"
  >
    <div
      className="group inline-block relative cursor-pointer"
    >
      <svg
        className="text-gray-400 hover:text-[#3b5998]"
        fill="currentColor"
        height={24}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
        />
      </svg>
      <div
        className="inline-block absolute z-10 text-white bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in -bottom-4 h-3 w-3 rotate-45"
      />
      <div
        className="inline-block absolute z-10 text-white bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in -bottom-8 p-1 text-xs leading-snug}"
      >
        Facebookでシェア
      </div>
    </div>
  </button>
  <button
    aria-label="line"
    className="react-share__ShareButton"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "border": "none",
        "color": "inherit",
        "cursor": "pointer",
        "font": "inherit",
        "padding": 0,
      }
    }
  >
    <div
      className="group inline-block relative cursor-pointer"
    >
      <svg
        className="text-gray-400 hover:text-[#00B900]"
        fill="currentColor"
        height={24}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M272.1 204.2v71.1c0 1.8-1.4 3.2-3.2 3.2h-11.4c-1.1 0-2.1-.6-2.6-1.3l-32.6-44v42.2c0 1.8-1.4 3.2-3.2 3.2h-11.4c-1.8 0-3.2-1.4-3.2-3.2v-71.1c0-1.8 1.4-3.2 3.2-3.2H219c1 0 2.1.5 2.6 1.4l32.6 44v-42.2c0-1.8 1.4-3.2 3.2-3.2h11.4c1.8-.1 3.3 1.4 3.3 3.1zm-82-3.2h-11.4c-1.8 0-3.2 1.4-3.2 3.2v71.1c0 1.8 1.4 3.2 3.2 3.2h11.4c1.8 0 3.2-1.4 3.2-3.2v-71.1c0-1.7-1.4-3.2-3.2-3.2zm-27.5 59.6h-31.1v-56.4c0-1.8-1.4-3.2-3.2-3.2h-11.4c-1.8 0-3.2 1.4-3.2 3.2v71.1c0 .9.3 1.6.9 2.2.6.5 1.3.9 2.2.9h45.7c1.8 0 3.2-1.4 3.2-3.2v-11.4c0-1.7-1.4-3.2-3.1-3.2zM332.1 201h-45.7c-1.7 0-3.2 1.4-3.2 3.2v71.1c0 1.7 1.4 3.2 3.2 3.2h45.7c1.8 0 3.2-1.4 3.2-3.2v-11.4c0-1.8-1.4-3.2-3.2-3.2H301v-12h31.1c1.8 0 3.2-1.4 3.2-3.2V234c0-1.8-1.4-3.2-3.2-3.2H301v-12h31.1c1.8 0 3.2-1.4 3.2-3.2v-11.4c-.1-1.7-1.5-3.2-3.2-3.2zM448 113.7V399c-.1 44.8-36.8 81.1-81.7 81H81c-44.8-.1-81.1-36.9-81-81.7V113c.1-44.8 36.9-81.1 81.7-81H367c44.8.1 81.1 36.8 81 81.7zm-61.6 122.6c0-73-73.2-132.4-163.1-132.4-89.9 0-163.1 59.4-163.1 132.4 0 65.4 58 120.2 136.4 130.6 19.1 4.1 16.9 11.1 12.6 36.8-.7 4.1-3.3 16.1 14.1 8.8 17.4-7.3 93.9-55.3 128.2-94.7 23.6-26 34.9-52.3 34.9-81.5z"
        />
      </svg>
      <div
        className="inline-block absolute z-10 text-white bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in -bottom-4 h-3 w-3 rotate-45"
      />
      <div
        className="inline-block absolute z-10 text-white bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in -bottom-8 p-1 text-xs leading-snug}"
      >
        LINEでシェア
      </div>
    </div>
  </button>
  <button
    aria-label="pocket"
    className="react-share__ShareButton"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "border": "none",
        "color": "inherit",
        "cursor": "pointer",
        "font": "inherit",
        "padding": 0,
      }
    }
  >
    <div
      className="group inline-block relative cursor-pointer"
    >
      <svg
        className="text-gray-400 hover:text-[#ee4056]"
        fill="currentColor"
        height={24}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M407.6 64h-367C18.5 64 0 82.5 0 104.6v135.2C0 364.5 99.7 464 224.2 464c124 0 223.8-99.5 223.8-224.2V104.6c0-22.4-17.7-40.6-40.4-40.6zm-162 268.5c-12.4 11.8-31.4 11.1-42.4 0C89.5 223.6 88.3 227.4 88.3 209.3c0-16.9 13.8-30.7 30.7-30.7 17 0 16.1 3.8 105.2 89.3 90.6-86.9 88.6-89.3 105.5-89.3 16.9 0 30.7 13.8 30.7 30.7 0 17.8-2.9 15.7-114.8 123.2z"
        />
      </svg>
      <div
        className="inline-block absolute z-10 text-white bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in -bottom-4 h-3 w-3 rotate-45"
      />
      <div
        className="inline-block absolute z-10 text-white bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in -bottom-8 p-1 text-xs leading-snug}"
      >
        Pocketに保存
      </div>
    </div>
  </button>
  <button
    aria-label="hatena"
    className="react-share__ShareButton"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "border": "none",
        "color": "inherit",
        "cursor": "pointer",
        "font": "inherit",
        "padding": 0,
      }
    }
  >
    <div
      className="group inline-block relative cursor-pointer"
    >
      <svg
        className="text-gray-400 hover:text-[#00A4DE]"
        fill="currentColor"
        height={24}
        role="img"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <title />
        <path
          d="M20.47 0C22.42 0 24 1.58 24 3.53v16.94c0 1.95-1.58 3.53-3.53 3.53H3.53C1.58 24 0 22.42 0 20.47V3.53C0 1.58 1.58 0 3.53 0h16.94zm-3.705 14.47c-.78 0-1.41.63-1.41 1.41s.63 1.414 1.41 1.414 1.41-.645 1.41-1.425-.63-1.41-1.41-1.41zM8.61 17.247c1.2 0 2.056-.042 2.58-.12.526-.084.976-.222 1.32-.412.45-.232.78-.564 1.02-.99s.36-.915.36-1.48c0-.78-.21-1.403-.63-1.87-.42-.48-.99-.734-1.74-.794.66-.18 1.156-.45 1.456-.81.315-.344.465-.824.465-1.424 0-.48-.103-.885-.3-1.26-.21-.36-.493-.645-.883-.87-.345-.195-.735-.315-1.215-.405-.464-.074-1.29-.12-2.474-.12H5.654v10.486H8.61zm.736-4.185c.705 0 1.185.088 1.44.262.27.18.39.495.39.93 0 .405-.135.69-.42.855-.27.18-.765.254-1.44.254H8.31v-2.297h1.05zm8.656.706v-7.06h-2.46v7.06H18zM8.925 9.08c.71 0 1.185.08 1.432.24.245.16.367.435.367.83 0 .38-.13.646-.39.804-.265.154-.747.232-1.452.232h-.57V9.08h.615z"
        />
      </svg>
      <div
        className="inline-block absolute z-10 text-white bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in -bottom-4 h-3 w-3 rotate-45"
      />
      <div
        className="inline-block absolute z-10 text-white bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in -bottom-8 p-1 text-xs leading-snug}"
      >
        はてなブックマークでシェア
      </div>
    </div>
  </button>
</div>
`;
